Assets/UnitTests/PlayModeTests/CalibrationTests.cs

```diff
@@ -687,6 +696,26 @@ public IEnumerator FocusOnPois_CheckLocalScales_z()
             }
         }
 
+        [UnityTest, Order(10)]
```

Comment:
I don't think a test is needed to check what text is written on the button.

Reply:
Hi @cawke  This is a regression problem. At some point this text was added and then, for some reason, returned to the 'OK' value. (I understand the test for 'OK' is 'Close' for IOS, Android and Windows). So using TDD principles, I added a test for this. Is this argument convincing to you? Thanks.

Reply:
Honestly, these seem to me arguments to keep the test, instead of deleting the test. If you make a change in the software and you don't change the set of tests, a test should fail. Tests are there to detect changes that could inadvertently break the software. I think, though, that the place I wrote the test in is wrong. Would we find an agreement in relocating the test to a suite of its own? Thanks.

Reply:
I still think that this test is only an inconvenience, no matter where it is located. We should cover more complicated things with tests, not like button text.

@Alvareon and @BenediktHensen please give us your opinion.

Reply:
Well, in my opinion, we shouldn't cover texts by tests. Our project has too many assets with texts. It doesn't really matter. We need to focus on the functionality.

Reply:
@Alvareon . I submit regressions cost time, and once you have learned how to write a test for one item of text, it is immediate to apply that knowledge for a new element. In addition, if I may, this documents the structure of our project. Thanks.

Reply:
My complaints about the test are mainly about how you find the prefab (direct path to the file) and how you find the component (by name throughout the prefab)

![image](https://user-images.githubusercontent.com/20320355/180446503-ffabded1-d587-4de9-961b-edaf7e46228b.png)

![image](https://user-images.githubusercontent.com/20320355/180446547-5f4b6d1f-f5c0-4df3-823f-e47f79419fe3.png)

Reply:
@cawke I understand that the path is relative and does not change, and so it is about the tag of the component. I could think of other ways of retrieving the prefab (... suggestions?). We don't have a table for the tags used for the elements of the prefab, I submit adding this table is the solution to make this test more architecture agnostic.

Reply:
I would also say that we should focus on testing the functionality. Ideally, there should be a separation between the functional layer and the visuals of the scene. So, the methods that we test are separated from the actual GameObjects, scene structure, prefabs, etc. This way, the method will work with just a minimal setup and there is no need to replicate the scene structure by loading prefabs.

The main trouble with such tests that require visual elements is the fact that we need to generate these visual scenes from the test's code. Here, retrieving assets from paths both on disk but also in the Unity scene can break the test easily. The same is true for some of the existing tests which start re-creating scene structure in the initialization code. If we move around some files, rename folders or GameObjects, this will break the test, although the functionality that the test checks would still work.

Reply:
@BenediktHensen  I understand your concerns, and they are reasonable. A test is meant to detect a problem, though, not necessarily to last for years. The solution, I think, is to use a Localisation table and verify the constant we use for the component. This way we decouple the component and the text, and we verify the text only. There is still the issue of some person inadvertently changing the reference in the component and I am unsure if the team wants to introduce localisation tables at this point, though.

Reply:
A test that works is more than nothing.

Reply:
Final though (I see this is already resolved): such test could be useful in the context of localisation - when we have language support for different languages (to check whether all or at least key items are actually updated when the language is switched)?